### DISCUSSION TITLE
Catch rate limit event and block client

### DISCUSSION THREAD

**Discussion by greenenvy** [emqx/emqx#8880] [Q&A]
Hello everyone,

So i was searching for any kind of rate limiting for MQTT brokers and i've come to EMQX, that has rate limiting in the configuration options.

The case i want to use rate limiting is about our clients IoT devices that have from time to time a faulty sensor which leads to massive spamming from a single device, and our need is to somehow block off that device.

From the docs i've learned that EMQX supports web hooks, and also has a HTTP API that allows some devices (per clientId, username or ip address) to be added to a blacklist.

My only concern from what i have tried is that there is no event for rate limiting to be catched with a web hook?

I've tried this on the open-source version 4.4.8 without success. And i'm not sure if my idea is even doable. Any way of blocking client when exceeding rate limit is ok.

Best regards, Stefan.

**Comment by lafirest** [comment]
Hello,  @greenenvy, yes, there is currently no rate-limiting related events, but on 4.4.x you can set the rate of publishing per client (session), so that although you can't block the device, you can control its side effects.

The idea that blocking or kicking the overloaded session is a good input, thanks.

**Reply by greenenvy** [reply]
@lafirest Is it to expect that this may be available in near future? It is only needed to have a event where you can attach a web hook, and then you can do the blacklist through the HTTP API.

I'm looking for this kind of behavior on any MQTT broker, and Emqx comes the closest to it.

**Reply by lafirest** [reply]
I think this would be a nice feature, we'll discuss it
